pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -909,6 +909,8 @@ func applyToTargetCluster(
 	applyFn func(obj []byte) (resource.ApplyResult, error),
 	logger log.FieldLogger,
 ) error {
+	logger = logger.WithField("uid", obj.GetUID()).WithField("resource_version", obj.GetResourceVersion())
```

Comment:
This will statically assign these field values at this point in time. We want to retrieve the values on the updated obj after the apply. So we need to explicitly add them with each log directive.